taf/tests/test_updater/update_utils.py

```diff
@@ -256,3 +257,112 @@ def _update_expect_error():
+
+        if client_commit != origin_commit:
+            successful_update = False
+            break
+
+    if successful_update:
+        check_if_commits_match(client_target_repos, auth_repo.path.parent.parent)
+    else:
+        for repo_name, client_repo in client_target_repos.items():
+            origin_repo = GitRepository(auth_repo.path.parent.parent, repo_name)
+            client_commit = client_repo.head_commit_sha()
+
+            # Get the target file path in the client repo
+            target_file_path = client_repo.path / "test1.txt"
+
+            # Ensure the target file exists
+            if not target_file_path.is_file():
+                raise FileNotFoundError(
+                    f"Target file {target_file_path} does not exist"
+                )
+
+            # Extract commit SHA from the target file in the client repo
+            target_commit_sha = client_repo._git(
+                f'log -n 1 --pretty=format:"%H" -- {target_file_path}'
+            ).strip('"')
+
+            # Assert that the top commits of target repositories are the same as the commit SHA specified in the corresponding target files
+            assert (
+                client_commit == target_commit_sha
+            ), f"Target repo {repo_name} should have the same top commit as specified in the corresponding target file"
+            assert (
+                client_commit != origin_commit
+            ), f"Target repo {repo_name} should not have the same commit as specified in the client's auth repo"
+
+
+def verify_partial_update(
+    client_dir: Path, origin_auth_repo: AuthenticationRepository, original_commits: dict
+):
+    """
+    Verify that the client's repositories are in the correct state following a partial update.
+    This means that the top commits of target repositories are the same as the top commits of origin repositories,
+    but not the same as the commits specified in the client's auth repo.
+    """
+    client_auth_repo = AuthenticationRepository(path=client_dir / origin_auth_repo.name)
+    client_target_repos = load_target_repositories(
+        origin_auth_repo, library_dir=client_dir
+    )
+
+    # Ensure the last validated commit exists in client's auth repo
+    check_last_validated_commit(client_auth_repo.path)
+
+    for repo_name, client_repo in client_target_repos.items():
+        origin_repo = GitRepository(origin_auth_repo.path.parent.parent, repo_name)
+        client_commit = client_repo.head_commit_sha()
+        origin_commit = origin_repo.head_commit_sha()
+
+        # Assert that the top commits of target repositories are the same as the top commits of origin repositories
+        assert (
+            client_commit == origin_commit
+        ), f"Target repo {repo_name} should have the same top commit as the origin repo"
+
+        # Assert that the top commits of target repositories are not the same as the commit specified in the client's auth repo
```

Comment:
To check if top commits or target repositories are the same as the commits specified in the auth repo, read the expected commit shas from target files: `
```
  target = client_auth_repo.get_target(repo_name)
  commit_sha = target["commit"]
  assert commit_sha ==  client_commit
```